core/server/master/src/main/java/alluxio/master/file/loadmanager/LoadManager.java

```diff
@@ -0,0 +1,310 @@
+  public CloseableIterator<Journal.JournalEntry> getJournalEntryIterator() {
+    return CloseableIterator.noopCloseable(mLoadPathToInfo.keySet().stream()
+            .map(loadInfo -> Journal.JournalEntry.newBuilder().setLoadDirectory(
+                            File.LoadDirectory.newBuilder()
+                                    .setSrcFilePath(loadInfo)
+                                    .setLoadId(0)
+                    )
+                    .build())
+            .iterator());
+  }
+
+  @Override
+  public boolean processJournalEntry(Journal.JournalEntry entry) {
+    if (entry.hasLoadDirectory()) {
+      File.LoadDirectory loadDirectory = entry.getLoadDirectory();
+      mLoadPathToInfo.put(loadDirectory.getSrcFilePath(),
+              new LoadInfo(mJobIdGenerator.getNewJobId(), loadDirectory.getSrcFilePath(),
+                      loadDirectory.getOptions().getBandWidth()));
+      return true;
+    } else {
+      return false;
+    }
+  }
+
+  @Override
+  public void resetState() {
+    mLoadPathToInfo.clear();
+  }
+
+  @Override
+  public CheckpointName getCheckpointName() {
+    return CheckpointName.LOAD_DIRECTORY;
+  }
+
+  /**
+   * Check load validation status, perform loadInfo option update accordingly.
+   * @param path the file path
+   * @param options the load options
+   * @return validation status
+   */
+  public ValidationStatus getValidationStatus(String path, LoadOptions options) {
+    if (mLoadPathToInfo.containsKey(path)) {
+      boolean isNewOption = mLoadPathToInfo.get(path).getLoadOptions().equals(options);
+      if (isNewOption) {
+        LoadInfo info = mLoadPathToInfo.get(path);
+        /*Only update bandwidth here.*/
+        long newBandWidth = options.getBandwidth();
+        info.getLoadOptions().setBandwidth(newBandWidth);
+        return ValidationStatus.Update_Options;
+      }
+      return ValidationStatus.Ignored;
+    }
+    return ValidationStatus.New;
+  }
+
+  static class Scheduler {
+    private static final int CAPACITY = 100;
+    private static final long TIMEOUT = 100;
+    private final ExecutorService mExecutorService = Executors.newSingleThreadExecutor();
+    private final BlockingQueue<Load> mLoadQueue = new LinkedBlockingQueue<>(CAPACITY);
+    private final AtomicInteger mCurrentSize = new AtomicInteger();
+    private final AtomicLong mIdGenerator = new AtomicLong();
+
+    void schedule(Load load)
+            throws ResourceExhaustedException, InterruptedException {
+      if (mCurrentSize.get() == CAPACITY) {
+        throw new ResourceExhaustedException(
+                "Insufficient capacity to enqueue load tasks!");
+      }
+
+      boolean offered = mLoadQueue.offer(load, TIMEOUT, TimeUnit.MILLISECONDS);
+      if (offered) {
+        mCurrentSize.incrementAndGet();
+      } else {
+        LOG.warn("Cannot enqueue load to the queue, may lose track on this load!"
+                + load.getDetailedInfo());
+      }
+
+      mExecutorService.submit(() -> {
```

Comment:
This should only be submitted if offered is true right?

Reply:
yeah, will add, thanks

Reply:
Actually i was thinking this capacity and queue is kind of "soft" capacity enforcement, because we will apply back pressure based on workers to control the flow for scheduling, this named capacity and queue may not be that meaningful. For now i will leave the capacity and queue there, and move this logic to inside `when offered is true`